File: case-study/src/service/CustomerService.js
import axios from "axios";

const findAll = async () => {
    try {
        const result = await axios.get(`http://localhost:7000/customer`)
        return result.data;
    } catch (e) {
        console.log(e)
    }
}
const findById = async (id) => {
    try {
        const result = await axios.get(`http://localhost:7000/customer/${id}`)
        return result.data;
    } catch (e) {
        console.log(e)
    }
}
const customerTypeList = async () => {
    try {
        const result = await axios.get(`http://localhost:7000/customerType`)
        return result.data;
    } catch (e) {
        console.log(e)
    }
}
const searchCustomer = async (name,customerType) => {
    try {
        const result = await axios.get(`http://localhost:7000/customer?name_like=${name}&customerType=${customerType}`)
        console.log(result)
        return result.data;
    } catch (e) {
        console.log(e)
    }
}
const deleteCustomer = async  (id) => {
    try {
        return await axios.delete(`http://localhost:7000/customer/${id}`)
    } catch (e) {
        console.log(e)
    }

}
const addCustomer = async (values) => {
    try {
        const res = axios.post(`http://localhost:7000/customer`,{...values})
        alert("Tạo thành công")
        return (await res).data;
    } catch (e) {
        console.log(e)
    }
}
const editCustomer = async (values) => {
    try {
        const res = axios.put(`http://localhost:7000/customer/${values.id}`,{...values})
        alert("Sửa thành công")
        return (await res).data;
    } catch (e) {
        console.log(e)
    }
}

export const customerService = {
    findAll,
    findById,
    deleteCustomer,
    addCustomer,
    customerTypeList,
    editCustomer,
    searchCustomer
}

File: demo1/src/main/java/com/example/demo/service/impl/CustomerService.java
package com.example.demo.service.impl;


import com.example.demo.dto.CustomerDTO;
import com.example.demo.model.Customer;
import com.example.demo.model.CustomerType;
import com.example.demo.repository.ICustomerRepository;
import com.example.demo.service.ICustomerService;
import org.springframework.beans.BeanUtils;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.PageImpl;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;


@Service
public class CustomerService implements ICustomerService {
    @Autowired
    private ICustomerRepository iCustomerRepository;
    @Autowired
    private ICustomerService customerTypeService;


    @Override
    public List<Customer> findAll() {
        return iCustomerRepository.findAll();
    }


    @Override
    public Customer findById(long id) {
        return iCustomerRepository.findById(id).get();
    }

    @Override
    public Customer findByCustomer(String name) {
        return iCustomerRepository.findByCustomer(name);
    }

    @Override
    public void create(Customer customer) {
        iCustomerRepository.save(customer);
    }

    @Override
    public void delete(Customer customer) {
        iCustomerRepository.delete(customer);
    }

    @Override
    public void save(Customer customer) {
        iCustomerRepository.save(customer);
    }


}

